keps/sig-node/4817-resource-claim-device-status/README.md

```diff
@@ -0,0 +1,731 @@
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This proposal enhances the `ResourceClaim.Status` by adding a new field
+`Devices`. The new field allows drivers to report driver-specific device
+status data for each allocated device in a resource claim. Allowing the
+drivers to report the device statuses will improve both observability and
+troubleshooting as well as enabling new functionalities such as, for example,
+if the IPs of a network device are reported, network services.
+
+This extension also establishes a standardization for specific type device data, 
+such as, for example, networking devices information.
+
+## Motivation
+
+As of now, when a device is configured in a pod/container, the state and
+characteristics of the device set during the configuration stage are invisible.
+The lack of this information is then a challenge for the users and developers
+to diagnose issues, verify configurations, and integrate the allocated
+resources into higher-level services. Reporting this information is then
+crucial in environments where device specific configurations are necessary.
+
+For certain types of devices, such as network interfaces, knowing the detailed
+characteristics of what has been allocated is particularly useful and crucial.
+For example, reporting the interface Name, MAC address and IP addresses of
+network interfaces in the status of a ResourceClaim can significantly help in
+configuring and managing network services, as well as in debugging network
+related issues. By including such device specific information, this proposal
+addresses existing gaps in visibility and facilitates better integration and
+management of resources.
+
+### Goals
+
+* Allow arbitrary, driver-specific information to be reported from the DRA
+  drivers for each allocated device in a ResourceClaim.
+* Establish a standardization for device status (e.g. Network Devices status).
+* Enable 3rd party implementation of new functionalities based on the Device
+  Status (e.g. Secondary Network Services if the IPs of a network device are
+reported).
+
+### Non-Goals
+
+* Implement new functionalities based on the Device Status (e.g. Kubernetes
+  EndpointSlice Controller supporting IPs reported by networking DRA drivers).
+* Modifying the Resource Claim workflow to include the device status report.
+
+## Proposal
+
+### API - ResourceClaim.Status
```

Comment:
The difference between "Proposal" and "Design Details" has always been confusing to me, but in this case it's especially confusing that the details of the proposal appear before the user stories, so I think this should mostly move to "Design Details".